index.bs

```diff
@@ -215,33 +215,21 @@ Request processing {#request-processing}
 
 When asked to <dfn abstract-op>append client hints to request</dfn> with |request| as input, run the following steps:
 
-<p>For each <a for=/>header</a> <a for=header>name</a> (<var>hintName</var>) in the registry's [=low entropy hint table=],
-if <var>request</var>'s <a for=request>header list</a>
-<a for="header list">does not contain</a> <var>hintName</var>, then
-<a for="header list">append</a>
-<var>hintName</var>/the corresponding value to
-<var>request</var>'s <a for=request>header list</a>.
-
-<p>If <var>request</var>'s <a for="request">client</a> is not null,
-  then <a for=list>for each</a> [=client hints token=] <var>hintName</var>
-  of <var>request</var>'s <a for="request">client</a>'s
-  <a for="environment settings object">client hints set</a>:
+Let |hintSet| be an empty [=client hints set=].
 
-<ol>
-  <li>
-   <p>Let <var>value</var> be the return value of running <a>find client hint value</a>, given <var>hintName</var> as input.
-
-
-  <li><p>If <var>request</var> is a <a>subresource request</a> and the result of running
-   <a href="https://w3c.github.io/webappsec-permissions-policy/#algo-should-request-be-allowed-to-use-feature">Should request be allowed to use feature?</a>,
-   given <var>request</var> and <var>hintName</var>’s
-   <a href="#policy-controlled-features">associated policy-controlled feature</a>, returns
-   <code>false</code>, then skip the next steps and continue to the next <var>hintName</var>.
-   [[!PERMISSIONS-POLICY]]
-  <li><p>If <var>request</var>'s <a for=request>header list</a> <a for="header list">does not
-  contain</a> <var>hintName</var>, a user agent should <a for="header list">append</a>
-  <var>hintName</var>/<var>value</var> to <var>request</var>'s <a for=request>header list</a>.
-</ol>
+For each [=client hints token=] in the registry's [=low entropy hint table=] (|lowEntropyHint|), [=set/append=] |lowEntropyHint| to |hintSet|.
```

Comment:
I don't think this will actually work. I think you want to:
* add the low entropy hints to a |hint set|
* Add the hints from the client hint set to |hint set|
* If client is not null, run the policy check on each hint in |hint set| and remove them from the set if they fail
* Add all the hints in the |hint set| as request headers

Reply:
If the client's hint is null, there's no client hint set to speak of, right?

Reply:
Yeah, the second step is also bound to the non null check

Reply:
I see what you're saying now.

Take a look